hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHATransactionBufferMonitorTask.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.hdds.scm.ha;
+
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A background service running in SCM to check and flush the HA Transaction
+ * buffer.
+ */
+public class SCMHATransactionBufferMonitorTask implements Runnable {
+  public static final Logger LOG =
+      LoggerFactory.getLogger(SCMHATransactionBufferMonitorTask.class);
+  private SCMRatisServer server;
+  private SCMHADBTransactionBuffer transactionBuffer;
+  private long flushInterval = 0;
+
+  /**
+   * SCMService related variables.
+   */
+
+  @SuppressWarnings("parameternumber")
+  public SCMHATransactionBufferMonitorTask(
+      SCMHADBTransactionBuffer transactionBuffer,
+      SCMRatisServer server, long flushInterval) {
+    this.flushInterval = flushInterval;
+    this.transactionBuffer = transactionBuffer;
+    this.server = server;
+  }
+
+  @Override
+  public void run() {
+    if (transactionBuffer.shouldFlush(flushInterval)) {
+      LOG.debug("Running TransactionFlushTask");
+      // set latest snapshot to null for force snapshot
+      // the value will be reset again when snapshot is taken
+      transactionBuffer.setLatestSnapshot(null);
```

Comment:
Is this necessary? What will happen if we do not set null

Reply:
@Xushaohong Thanks for review, above is done for full snapshot as,

- Ratis will try to get diff of current applied Index and latest snapshot Index, with minGapValue (default value is 1024), if its less than do not take snapshot
- But since this is force snapshot, and setting LatestSnapshot to null, its value is treated as "0", so diff after sometime crosses 1024 value and snapshot is taken.
- Further reset is not required with old value as when taking snapshot, it will set with latest applied index. (I will add if value is not set during snapshot due to any failure for this case).

Note: At least 1024 transaction is required in this case also, as this limitation should be ok after discussion with Nicholas,

@szetszwo Please share if this logic is fine as discussed.